src/unit-stack-configuration-workspace/unit-stack-configuration-checks.service.ts

```diff
@@ -60,20 +69,16 @@ export class UnitStackConfigurationChecksService {
       ) {
         // Check the end date if the record exists.
         errorList.push(
-          `The End Date of the Unit Stack Configuration cannot be before the end date of the last Emission Evaluation for the Unit or Stack/Pipe`,
+          'The End Date of the Unit Stack Configuration cannot be before the end date of the last Emission Evaluation for the Unit or Stack/Pipe.',
         );
-      } else if (!uscRecord) {
+      } else if (
+        !uscRecord &&
+        new Date(usc.beginDate) <= new Date(evaluation.reportingPeriod.endDate)
```

Comment:
Does this handle the Begin Date being null?

Reply:
Begin Date is checked in the DTO to be a valid date.

Reply:
👍